docs/reference/ros.md

```diff
@@ -0,0 +1,89 @@
+# Robot Operating System
+
+Robot Operating System (ROS) is a set of software libraries and tools for building robot applications [^1].
+It provides functionality for hardware abstraction, device drivers, communication between processes over
+multiple machines, tools for testing and visualization, and much more [^2].
+
+We uses ROS because it is open-source, language-agnostic, and built with cross-collaboration in mind.
+It enables our subteams to work independently on well-defined components of our software system,
+without having to worry about the hardware it runs on or the implementation of other components.
+
+[The official ROS 2 documentation](https://docs.ros.org/en/eloquent/index.html) contains everything you need
+to get started using ROS. From it we have hand-picked the resources that are most relevant to our current and expected
+future usage of ROS assuming that you use [our preconfigured workspace](https://github.com/UBCSailbot/sailbot_workspace).
+Installing and setting up ROS without our workspace is outside the scope of this site.
+
+## Workspace Configuration
+
+To get our workspace configuration running on your computer:
+
+1. Set it up by following the instructions in [its README](https://github.com/UBCSailbot/sailbot_workspace#setup)
+2. Uncomment the ROS 2 tutorials section in [`.devcontainer/Dockerfile`](https://github.com/UBCSailbot/sailbot_workspace/blob/main/.devcontainer/Dockerfile),
+   then run the "Remote-Containers: Rebuild Container" VS Code command, to install the tutorials' dependencies
+3. Uncomment the ROS 2 tutorials section in [`src/new_project.repos`](https://github.com/UBCSailbot/sailbot_workspace/blob/main/src/new_project.repos),
+   then run the "setup" VS Code task, to clone the repositories used in the tutorials
+
+Our workspace configuration contains easier methods of accomplishing some of the tutorial steps, or eliminates the need
+for them altogether.
```

Comment:
Any examples of what functionalities are accomplished using the workspace? If there are not too many, a quick exhaustive list would be nice.